### PR TITLE
Add mtbseq workflow

### PR DESCRIPTION
**Pull request by Mxrcon**
# MTBseq  addition

Hey there :wave:,

This PR adds 
- all required mtbseq modules
- mtbseq workflows and sub workflows
- parameters base configuration as used in `mtbseq_nf`
- test profiles for MTBseq

the module structure follows:

```
── local
│   ├── mtbseq
│   │   ├── tbamend
│   │   │   ├── environment.yml
│   │   │   ├── main.nf
│   │   │   └── meta.yml
│   │   ├── tbbwa
│   │   │   ├── environment.yml
│   │   │   ├── main.nf
│   │   │   └── meta.yml
│   │   ├── tbfull
│   │   │   ├── environment.yml
│   │   │   ├── main.nf
│   │   │   └── meta.yml
│   │   ├── tbgroups
│   │   │   ├── environment.yml
│   │   │   ├── main.nf
│   │   │   └── meta.yml
│   │   ├── tbjoin
│   │   │   ├── environment.yml
│   │   │   ├── main.nf
│   │   │   └── meta.yml
│   │   ├── tblist
│   │   │   ├── environment.yml
│   │   │   ├── main.nf
│   │   │   └── meta.yml
│   │   ├── tbpile
│   │   │   ├── environment.yml
│   │   │   ├── main.nf
│   │   │   └── meta.yml
│   │   ├── tbrefine
│   │   │   ├── environment.yml
│   │   │   ├── main.nf
│   │   │   └── meta.yml
│   │   ├── tbstats
│   │   │   ├── environment.yml
│   │   │   ├── main.nf
│   │   │   └── meta.yml
│   │   ├── tbstrains
│   │   │   ├── environment.yml
│   │   │   ├── main.nf
│   │   │   └── meta.yml
│   │   └── tbvariants
│   │       ├── environment.yml
│   │       ├── main.nf
│   │       └── meta.yml
│   └── utils
│       └── rename_files.nf
```

I kindly request for your review @abhi18av and @ycode-sh for proceeding with this work.

## PR checklist

- [x] Perform tests on the workflow
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: conf/mtbseq-base.config [hunk 1]
```diff
@@ -0,0 +1,96 @@
+/*
```
Comment (abhi18av) [inline conf/mtbseq-base.config]: @Mxrcon , the file name should be with underscore `_`.

Comment (Mxrcon) [reply conf/mtbseq-base.config]: good catch! I'll rename it!

**Comment by abhi18av** [comment]
@Mxrcon the major problem here is related to `nf-validation` plugin.

In the `MTBseq-nf` when we initiated the `nf-schema` plugin was fine but now it has been deprecated in favor of `nf-validation`.

[plugin_includes](https://nf-co.re/tools/docs/3.0.2/pipeline_lint_tests/plugin_includes) - Found a nf-validation plugin import in ome/runner/work/tbanalyzer/tbanalyzer/workflows/mtbseqnf.nf, but nf-schema was used in nextflow.config

**Comment by Mxrcon** [comment]
> @Mxrcon the major problem here is related to `nf-validation` plugin.
> 
> In the `MTBseq-nf` when we initiated the `nf-schema` plugin was fine but now it has been deprecated in favor of `nf-validation`.
> 
> [plugin_includes](https://nf-co.re/tools/docs/3.0.2/pipeline_lint_tests/plugin_includes) - Found a nf-validation plugin import in ome/runner/work/tbanalyzer/tbanalyzer/workflows/mtbseqnf.nf, but nf-schema was used in nextflow.config

Hmm, just checked it, the port from `nf-validation` to `nf-schema` is very straightfoward, just added it and I'll now work into the test of the mtbseq workflow

**Comment by Mxrcon** [comment]
# Update after Tests
Hey, I've performed local tests for the latest commits and here are the outputs:

## Command executed: 
`nextflow run main.nf -profile conda -c mtbseq-test.conf --mode mtbseq -resume --mtbseq_parallel`
```
executor >  local (27)
[89/24bbfd] process > NFCORE_TBANALYZER:MTBSEQ_NF:QUALITY_CHECK:RENAME_FILES (4781-04) [100%] 3 of 3 ✔
[66/e79bfd] process > NFCORE_TBANALYZER:MTBSEQ_NF:QUALITY_CHECK:FASTQC (4781-04)       [100%] 3 of 3 ✔
[c7/9408b0] process > NFCORE_TBANALYZER:MTBSEQ_NF:PARALLEL_MODE:TBBWA (4781-04)        [100%] 3 of 3 ✔
[3a/38d282] process > NFCORE_TBANALYZER:MTBSEQ_NF:PARALLEL_MODE:TBREFINE (4781-04)     [100%] 3 of 3 ✔
[67/526446] process > NFCORE_TBANALYZER:MTBSEQ_NF:PARALLEL_MODE:TBPILE (4781-04)       [100%] 3 of 3 ✔
[5f/05673e] process > NFCORE_TBANALYZER:MTBSEQ_NF:PARALLEL_MODE:TBLIST (4781-04)       [100%] 3 of 3 ✔
[49/2405d7] process > NFCORE_TBANALYZER:MTBSEQ_NF:PARALLEL_MODE:TBVARIANTS (4781-04)   [100%] 3 of 3 ✔
[8b/cb38bc] process > NFCORE_TBANALYZER:MTBSEQ_NF:PARALLEL_MODE:TBSTATS (cohort)       [100%] 1 of 1 ✔
[fc/6339ea] process > NFCORE_TBANALYZER:MTBSEQ_NF:PARALLEL_MODE:TBSTRAINS (cohort)     [100%] 1 of 1 ✔
[e1/c2aa8e] process > NFCORE_TBANALYZER:MTBSEQ_NF:PARALLEL_MODE:TBJOIN (cohort)        [100%] 1 of 1 ✔
[55/f73876] process > NFCORE_TBANALYZER:MTBSEQ_NF:PARALLEL_MODE:TBAMEND (cohort)       [100%] 1 of 1 ✔
[9d/39b91d] process > NFCORE_TBANALYZER:MTBSEQ_NF:PARALLEL_MODE:TBGROUPS (cohort)      [100%] 1 of 1 ✔
[00/a543e4] process > NFCORE_TBANALYZER:MTBSEQ_NF:MULTIQC                              [100%] 1 of 1 ✔
-[nf-core/tbanalyzer] Pipeline completed successfully-
Completed at: 28-Oct-2024 20:32:17
Duration    : 2h 16m 43s
CPU hours   : 6.6
Succeeded   : 27

```

## Command Executed: 
`nextflow run main.nf -profile conda -c mtbseq-test.conf --mode mtbseq -resume`
```
executor >  local (11)
[ea/1186c5] process > NFCORE_TBANALYZER:MTBSEQ_NF:QUALITY_CHECK:RENAME_FILES (3494-04) [100%] 3 of 3 ✔
[de/f3d790] process > NFCORE_TBANALYZER:MTBSEQ_NF:QUALITY_CHECK:FASTQC (3494-04)       [100%] 3 of 3 ✔
[24/281c56] process > NFCORE_TBANALYZER:MTBSEQ_NF:TBFULL (cohort)                      [100%] 1 of 1 ✔
[27/1965c8] process > NFCORE_TBANALYZER:MTBSEQ_NF:TBJOIN (cohort)                      [100%] 1 of 1 ✔
[6d/4c6c3b] process > NFCORE_TBANALYZER:MTBSEQ_NF:TBAMEND (cohort)                     [100%] 1 of 1 ✔
[de/d58589] process > NFCORE_TBANALYZER:MTBSEQ_NF:TBGROUPS (cohort)                    [100%] 1 of 1 ✔
[fe/859eb8] process > NFCORE_TBANALYZER:MTBSEQ_NF:MULTIQC                              [100%] 1 of 1 ✔
-[nf-core/tbanalyzer] Pipeline completed successfully-
Completed at: 28-Oct-2024 23:38:20
Duration    : 3h 5m 54s
CPU hours   : 3.0
Succeeded   : 11

```


Hope that this is enough for a merge, a I would like to proceed with pipeline documentation in another PR if possible.

@abhi18av, I would like to request a new review on this code before our next sync meet.

Kindly,
Davi

**Review by abhi18av — APPROVED** [review]
Hi @Mxrcon , thank you for accommodating the change request and also for the end-to-end test for the MTBseq `mode`. I am going ahead with the merge and then we can focus on local `MAGMA` modules.

**Comment by abhi18av** [comment]
I have just tried to change the `mode` to `method` parameter in https://github.com/nf-core/tbanalyzer/commit/47670645b33e632772efefb02423537212b4e016

**Pull request merged**